Add option to display dot files

ciscam opened the pull request:
According to https://github.com/DirectoryLister/DirectoryLister/discussions/701
Default behavior is not changed.

This new configuration item is used in FinderFactory, which now includes Finder.ignoreDotFiles() to the initialization.
Before, without setting this attribute, the class Finder would default to ignoreDotFiles TRUE.

Functionality:
* added hide_dot_files variable to config/app.php
* added hide_dot_files to FinderFactory initialization

Optional:
* added HIDE_DOT_FILES config line to .env.example for user-configurability
* added .env and .env.example to app_files (always hide). Those are configs only for internal use, right?

Tested by modifying the files of 3.8.3-release archive, with a file structure:
./.dot_file
./.dot_dir/
./.dot_dir/.dot_file

I did not look at the zip modules, as it seems to include whatever files/dirs Finder includes. Downloads of the directories contained dot files/dirs as expected.

PHLAK commented:
Thanks for the PR! Let's get those failing tests fixed and we can get this merged.

ciscam commented:
Tests should work now.
* Fixed interpunctional misunderstanding
* Added '.env' and '.env.example' to the expected hidden list containing app_files in HiddenFilesTest.php
* Added two tests for visibility and browseability of dot_files to FinderFactoryTest.php
